### PR TITLE
Add dependency update documentation

### PR DESCRIPTION
**Pull request by petrkotas**
### What this PR does / why we need it:

Add vendorind and dependency update documentation with
vendoring process of ARO-RP.

Signed-off-by: Petr Kotas <pkotas@redhat.com>

### Test plan for issue:

None, doc update.

### Is there any documentation that needs to be updated for this PR?

It is documentation itself.

### CODE REVIEW

File: docs/updating-dependencies.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# Updating dependencies aka vendoring
```
Comment (mjudeikis) [inline docs/updating-dependencies.md:1]: Match header with fine name 
```
# Updating dependencies
```

File: docs/updating-dependencies.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+To update golang dependencies one have to choose between two options:
```
Comment (jim-minter) [inline docs/updating-dependencies.md:3]:
```suggestion
To update golang dependencies one has to choose between two options:
```

File: docs/updating-dependencies.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+The reason for calling script instead of directly calling:
+
+```bash
+go get -u ./...
+go mod tidy
+go mod vendor
+```
+
+is that packages modified in this script does not fully support modules and
```
Comment (jim-minter) [inline docs/updating-dependencies.md:28]:
```suggestion
is that packages modified in this script do not fully support modules and
```

File: docs/updating-dependencies.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+semantic versioning via tags. Therefore the proper version parsed from the version
```
Comment (jim-minter) [inline docs/updating-dependencies.md:29]:
```suggestion
semantic versioning via tags. Therefore the proper version is parsed from the version
```

File: docs/updating-dependencies.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+branch and fixed using replace directive. Otherwise it will upgrade every time
+the command is started.
+
+When upgrading to never version of OpenShift, this script have to be updated to
```
Comment (jim-minter) [inline docs/updating-dependencies.md:33]:
```suggestion
When upgrading to a newer version of OpenShift, this script has to be updated to
```

File: docs/updating-dependencies.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+reflect the proper release.
+
+
+## Update only required dependencies
+
+When adding new dependencies or updating old ones, based by the requirement of
```
Comment (jim-minter) [inline docs/updating-dependencies.md:39]:
```suggestion
When adding new dependencies or updating old ones, based on the requirement of
```

File: docs/updating-dependencies.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+the the PR, one can simply call
```
Comment (jim-minter) [inline docs/updating-dependencies.md:40]:
```suggestion
the PR, one can simply call
```

File: docs/updating-dependencies.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+
+```bash
+go get <module to be downloaded> OR
+go get -u <module to be updated>
```
Comment (jim-minter) [inline docs/updating-dependencies.md:44]: mention the `go get -u <module>@<release>` syntax here?

File: docs/updating-dependencies.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+go mod tidy
+go mod vendor
+```
+
+vendor is required as ARO is still mirroring all dependencies locally for the CI
+and due to problematic OpenShift packages.
```
Comment (jim-minter) [inline docs/updating-dependencies.md:51]:
```suggestion
make vendor is required as ARO mirrors all dependencies locally for CI reliability and reproducability.
```

File: docs/updating-dependencies.md [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+
+**NOTE** that when running `go mod vendor` only modules that are used in the
+source code will be vendored in.
+
+**NOTE** when updating the package modified in `hack/update-go-module-dependencies.sh`
```
Comment (jim-minter) [inline docs/updating-dependencies.md:56]:
```suggestion
**NOTE** when updating a package modified in `hack/update-go-module-dependencies.sh`
```

**Review by jim-minter — CHANGES_REQUESTED** [review]
+1 with changes - thanks @petrkotas

**Comment by petrkotas** [comment]
Thank you both for the checks.

**Review by mjudeikis — APPROVED** [review]

**Pull request merged**